javacord-core/src/main/java/org/javacord/core/entity/message/MessageImpl.java

```diff
@@ -185,19 +190,25 @@ public MessageImpl(DiscordApiImpl api, TextChannel channel, JsonNode data) {
             }
         }
 
-        if (data.has("mention_roles") && !data.get("mention_roles").isNull()) {
+        if (data.hasNonNull("mention_roles")) {
             getServer().ifPresent(server -> {
                 for (JsonNode roleMentionJson : data.get("mention_roles")) {
                     server.getRoleById(roleMentionJson.asText()).ifPresent(roleMentions::add);
                 }
             });
         }
 
-        if (data.has("activity") && !data.get("activity").isNull()) {
+        if (data.hasNonNull("activity")) {
             activity = new MessageActivityImpl(this, data.get("activity"));
         } else {
             activity = null;
         }
+
+        if (data.has("nonce")) {
```

Comment:
Just one thing: Maybe better use 
```suggestion
        if (data.hasNonNull("nonce")) {
```
otherwise, a `null` nonce would appear at `"null"`. I'm not sure if Discord ever sends null-nounces, but it's Discord: You can never know.

Reply:
Changed it.